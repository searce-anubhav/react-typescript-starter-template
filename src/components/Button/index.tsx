import * as React from 'react';

interface ButtonProps {
	className: string;
}

const Button: React.FC<ButtonProps> = ({
	className,
	children,
}): JSX.Element => {
	return (
		<React.Fragment>
			<button className={className}>{children}</button>
		</React.Fragment>
	);
};

export default Button;
