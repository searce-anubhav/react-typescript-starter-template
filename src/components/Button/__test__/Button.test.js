import React from 'react';
import renderer from 'react-test-renderer';
import Button from 'components/Button';

test('Render Button Snapshot and Match', () => {
	const component = renderer.create(<Button>Hello</Button>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
