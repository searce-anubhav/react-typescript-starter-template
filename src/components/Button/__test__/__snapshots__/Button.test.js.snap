// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Snapshot and Match 1`] = `
<button>
  Hello
</button>
`;
